url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParser.java

```diff
@@ -0,0 +1,3669 @@
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+    jj_consume_token(91);
+         {if (true) return ins;}
+    throw new Error("Missing return statement in function");
+  }
+
+// DeleteStatement ::= DELETE [FROM] TableReference [WHERE SQLExpression]
+  static final public MDelete DeleteStatement() throws ParseException {
+  MDelete d;
+  MExp e;
+  String s;
+    jj_consume_token(K_DELETE);
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case K_FROM:
+      jj_consume_token(K_FROM);
+      break;
+    default:
+      jj_la1[25] = jj_gen;
+      ;
+    }
+    s = TableReference();
+                                             d = new MDelete(s);
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case K_WHERE:
+      jj_consume_token(K_WHERE);
+      e = SQLExpression();
+                                   d.addWhere(e);
+      break;
+    default:
+      jj_la1[26] = jj_gen;
+      ;
+    }
+    jj_consume_token(91);
+      {if (true) return d;}
+    throw new Error("Missing return statement in function");
+  }
+
+// QueryStatement ::= SelectStatement
+  static final public MQuery QueryStatement() throws ParseException {
+  MQuery q;
+    q = SelectStatement();
+    jj_consume_token(91);
+                                {if (true) return q;}
+    throw new Error("Missing return statement in function");
+  }
+
+/* ---------------- General Productions --------------------- */
+  static final public String TableColumn() throws ParseException {
+  StringBuffer buf = new StringBuffer();
+  String s;
+    // user.table.column
+        s = OracleObjectName();
+                             buf.append(s);
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case 93:
+      jj_consume_token(93);
+      s = OracleObjectName();
+                                   buf.append("." + s);
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case 93:
+        jj_consume_token(93);
+        s = OracleObjectName();
+                                   buf.append("." + s);
+        break;
+      default:
+        jj_la1[27] = jj_gen;
+        ;
+      }
+      break;
+    default:
+      jj_la1[28] = jj_gen;
+      ;
+    }
+      {if (true) return buf.toString();}
+    throw new Error("Missing return statement in function");
+  }
+
+  static final public String OracleObjectName() throws ParseException {
+  Token t;
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case S_IDENTIFIER:
+      t = jj_consume_token(S_IDENTIFIER);
+                         {if (true) return t.toString();}
+      break;
+    case S_QUOTED_IDENTIFIER:
+      t = jj_consume_token(S_QUOTED_IDENTIFIER);
+                                                                              {if (true) return t.toString();}
+      break;
+    default:
+      jj_la1[29] = jj_gen;
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+    throw new Error("Missing return statement in function");
+  }
+
+  static final public String Relop() throws ParseException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.